algorithms/bug2.py

```diff
@@ -0,0 +1,291 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+import os
+import sys
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+
+from tools.map import Map
+
+
+max_index = -1
+
+
+def reached_goal(grid, x, y):
+    '''
+    Checks if the robot reached the goal.
+    
+    Parameters
+    ----------
+    grid: 
+        2D map containing the robot, goal and obstacles.  
+    x: Numeric
+        Current x location of the robot.
+    y: Numeric
+        Current y location of the robot.
+        
+    Returns
+    -------
+    True: Boolean
+        Robot reached the goal.
+    False: Boolean
+        Robot did not reached the goal.
+    '''
+    if (grid[x][y] == 1):
+        return True
+    return False
+
+
+
+def if_boundary(curr_x, curr_y, grid, path):
```

Comment:
How do you deal with the situation where there are multiple obstacles? It appears that this will just recognize the first. This is a very uncommon situation, but it is something to think about how your entire algorithm handles that.